Bad _py.cpp code for templates with default arguments that are themselves templates

```c++
// @trick_parse{everything}
template <class T = int>
class Bar {}; 

template <class T = Bar<> >
class Foo {
    public:
    Foo() {}
};
```
```
build/home/test/sandbox/SIM_sandbox/models/Foo_py.cpp:3518:1: error: invalid use of template-name ‘Foo’ without an argument list
 Foo * castAsFoo( PyObject * obj1 ) {
 ^
build/home/test/sandbox/SIM_sandbox/models/Foo_py.cpp: In function ‘PyObject* _wrap_castAsFoo(PyObject*, PyObject*)’:
build/home/test/sandbox/SIM_sandbox/models/Foo_py.cpp:4389:7: error: missing template arguments before ‘*’ token
   Foo *result = 0 ;
```

The template-matching regular expression in [`convert_swig`](https://github.com/nasa/trick/blob/master/libexec/trick/convert_swig#L91) is too restrictive by not allowing nested `>` characters. We should instead use a non-greedy match anything `.*?`

--- a/include/trick/swig/cast_as.i
+++ b/include/trick/swig/cast_as.i
@@ -7,7 +7,7 @@
 %define %trick_cast_as(TYPE, NAME)
 
 %inline %{
-/* caseAs##NAME casts incoming ptrs as the target type.  Takes incoming object and tests if it
+/* castAs##NAME casts incoming ptrs as the target type.  Takes incoming object and tests if it
    is a compatible type, a void *, or an integer (raw pointer value).  If it is any of these
    types it returns the incoming value as a pointer to the outgoing type.  The integer
    conversion is used by the input processor to associate named allocations in the

--- a/libexec/trick/convert_swig
+++ b/libexec/trick/convert_swig
@@ -88,7 +88,7 @@ my $class_def   = qr/(?:class|struct)\s*                   # keyword class or st
                (?:\{|:(?!\:))                              # { or punctuator :
              /sx ;
 my $template_def   = qr/template\s*                        # keyword template
-            <[^>]+>\s*                                     # template parameters
+            <.*?>\s*                                       # template parameters
             (?:class|struct)                               # keyword class or struct
             \s+[_A-Za-z]\w*\s*                             # class name
                /sx ;

